src/index.js

```diff
@@ -75,4 +102,28 @@ export default class SaucelabsConnector {
 
         await closeTunnel();
     }
+
+    async waitForFreeMachines (machinesCount, requestInterval, attemptsCount) {
+        var attempt          = 1;
+        var isMachinesEnough = false;
+
+        while (!isMachinesEnough && attempt <= attemptsCount) {
+            var freeMachinesCount = await this._getFreeMachinesCount();
+
+            isMachinesEnough = await freeMachinesCount >= machinesCount;
+
+            if (!isMachinesEnough) {
+                process.stdout.write('The amount of free machines is below than requested.\n');
+                attempt++;
+
+                if (attempt <= attemptsCount)
+                    await wait(requestInterval);
+            }
+            else
+                process.stdout.write('The amount of free machines is enough for request.\n');

```

Comment:
I think this message is redundant. We should log about unsuccessful attempts only.